### DISCUSSION TITLE
ScrollView fails when contentInsetAdjustmentBehavior is automatic

### DISCUSSION THREAD

**Discussion by grahammendick** [reactwg/react-native-new-architecture#30] [Q&A] [answered]
I've started updating my navigation library to work with the new architecture and I've run into a problem on iOS. The `ScrollView` doesn't work when `contentInsetAdjustmentBehavior` is set to `automatic`. You can see what I mean in the video below. The first time you navigate to the second screen then the `ScrollView` behaves correctly. But the second time you navigate you can see that the `ScrollView` isn't adjusting for the navigation bar.

## Reproduction of the problem
I've created a [repository with a reproduction of the bug](https://github.com/grahammendick/content-inset-adjustment-auto-bug). The README in the repo contains the steps needed to recreate the problem. There are two apps in the repo, one built with the old architecture and the other built with the new architecture. You can see that the bug only happens on the new architecture. The only dependencies in the apps are the 3 packages that make up my navigation library.

https://user-images.githubusercontent.com/1761227/163725731-2bcbcae4-17a2-4ac1-9e52-cfced9a39c9c.mov

**Comment by sammy-SC** [comment]
Thank you for the report, this looks like a bug. It seems to happen when the same instance of scroll view is reused. 

@grahammendick would you be interested in creating a PR to fix this issue? I would be happy to provide code pointers.

**Reply by grahammendick** [reply]
I’m in the process of updating my Navigation router to the new architecture. I’ve already spent about 3 weeks on it and I’m not even a quarter of the way through. My library is 100% native - every component renders to its native counterpart - so the new architecture has given me a lot to rewrite and test.

Having said that, this scroll view bug is a showstopper for my Navigation router. I can’t upgrade to the new architecture without a fix for this. But I can see that, from the perspective of the React Native team, this bug might be low priority. So if the only way to ensure this gets fixed is for me to work on it then I will definitely make the time.

Please let me know what priority this bug is for the React Native team.

**Comment by dmytrorykun** [comment, marked as answer]
Hi @grahammendick, thank you for the reporting this bug and for the test project! We have landed a fix. It will be available in v0.69

**Reply by grahammendick** [reply]
Thank you v much! Out of interest could you point me at the fix please? I'm guessing you added something to the Scroll View's `prepareForRecycle`?

**Reply by dmytrorykun** [reply]
Yeah, you're right 🙂. Here it is https://github.com/facebook/react-native/commit/28a65f438789c29309d6e7c58063a73ca721ef43

**Reply by grahammendick** [reply]
Thanks 👍
